src/layout/Checkboxes/CheckboxesContainerComponent.tsx

```diff
@@ -101,26 +109,93 @@ export const CheckboxContainerComponent = ({
             id={`${id}-${option.label.replace(/\s/g, '-')}`}
-            name={option.value}
             key={option.value}
-            description={lang(option.description)}
+            description={lang(option.description) as string}
             value={option.value}
             checked={selected.includes(option.value)}
-            size='small'
-          >
-            {
-              <span className={cn({ 'sr-only': hideLabel }, classes.checkBoxLabelContainer)}>
-                {langAsString(option.label)}
-                {option.helpText && (
-                  <HelpText title={getPlainTextFromNode(option.helpText)}>{lang(option.helpText)}</HelpText>
-                )}
-              </span>
-            }
-          </Checkbox>
+            alertOnChange={alertOnChange}
+            renderedInTable={overrideDisplay?.renderedInTable}
+            singleCheckbox={calculatedOptions.length === 1}
+            label={option.label}
+            helpText={option.helpText}
+            handleChange={handleChange}
+            checkedItems={checkedItems}
+          />
         ))}
       </Checkbox.Group>
     </div>
   );
 };
+
+interface ICheckboxProps {
+  id: string;
+  description?: string;
+  value: string;
+  label: string;
+  helpText?: string;
+  checked: boolean;
+  alertOnChange?: boolean;
+  renderedInTable?: boolean;
+  singleCheckbox?: boolean;
+  handleChange: (checkedItems: string[], confirmedChange: boolean) => void;
+  checkedItems: string[];
+}
+
+const EnhancedCheckbox = ({
+  id,
+  description,
+  value,
+  label,
+  helpText,
+  checked,
+  alertOnChange,
+  renderedInTable,
+  singleCheckbox,
+  handleChange,
+  checkedItems,
+}: ICheckboxProps) => {
+  const [popoverOpen, setPopoverOpen] = React.useState(false);
+  const { lang, langAsString } = useLanguage();
+  const hideLabel = renderedInTable === true && singleCheckbox;
+
+  const handleUncheck = () => {
+    setPopoverOpen(false);
+    handleChange(checkedItems, true);
+  };
+  const checkBox = (
+    <Checkbox
+      id={id}
+      name={value}
+      description={description}
+      value={value}
+      checked={checked}
+      size='small'
+      onChange={alertOnChange && checked ? () => setPopoverOpen(true) : undefined}
+    >
+      {
+        <span className={cn({ 'sr-only': hideLabel }, classes.checkBoxLabelContainer)}>
+          {langAsString(label)}
+          {helpText && <HelpText title={getPlainTextFromNode(helpText)}>{lang(helpText)}</HelpText>}
+        </span>
+      }
+    </Checkbox>
+  );
+  if (popoverOpen) {
+    return (
+      <DeleteWarningPopover
+        trigger={checkBox}
+        deleteButtonText={lang('form_filler.alert_confirm') as string}
+        messageText={lang('form_filler.checkbox_alert') as string}
+        onCancelClick={() => {
+          setPopoverOpen(false);
+        }}
+        onPopoverDeleteClick={handleUncheck}
+        open={popoverOpen}
+        setOpen={setPopoverOpen}
+      />
+    );
+  }
```

Comment:
Why is this conditionally rendered on `popoverOpen` when the`DeleteWarningPopover` uses `popoverOpen` to say whether it is open?